cmd/opm/alpha/veneer/semver.go

```diff
@@ -18,17 +18,25 @@ import (
 func newSemverCmd() *cobra.Command {
 	output := ""
 	cmd := &cobra.Command{
-		Use:   "semver <filename>",
-		Short: "Generate a file-based catalog from a single 'semver veneer' file",
-		Long:  `Generate a file-based catalog from a single 'semver veneer' file`,
-		Args:  cobra.MinimumNArgs(1),
+		Use:   "semver [FILE]",
+		Short: "Generate a file-based catalog from a single 'semver veneer' file \nWhen FILE is '-' or not provided, the veneer is read from standard input",
+		Long:  "Generate a file-based catalog from a single 'semver veneer' file \nWhen FILE is '-' or not provided, the veneer is read from standard input",
+		Args:  cobra.MaximumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			ref := args[0]
-
 			var (
-				err error
+				data io.Reader
+				err  error
 			)
 
+			if len(args) == 0 || args[0] == "-" {
+				data = cmd.InOrStdin()
+			} else {
+				data, err = os.Open(args[0])
+				if err != nil {
+					return err
+				}
+			}
```

Comment:
Move this logic to it's own function that returns an io.Reader? It would help remove the if/else conditional structure that can help improve code readability.

Reply:
I'm worried that pulling that logic out into its own function will obfuscate things more because it seems like it'd be more confusing to have to find/look in a separate function to understand what the command's input argument logic is. I can add a comment explaining the logic to help improve readability of the if/else structure.